tests/scheduled_task/scheduled_task_suite_test.go

```diff
@@ -0,0 +1,237 @@
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/k8ssandra/cass-operator/tests/kustomize"
+	ginkgo_util "github.com/k8ssandra/cass-operator/tests/util/ginkgo"
+	"github.com/k8ssandra/cass-operator/tests/util/kubectl"
+)
+
+var (
+	testName              = "ScheduledTask"
+	namespace             = "test-scheduled-task"
+	dcName                = "dc1"
+	dcYaml                = "../testdata/default-scheduled-task-dc.yaml"
+	scheduledTaskYaml     = "../testdata/tasks/scheduled_cleanup_task.yaml"
+	scheduledTaskName     = "scheduled-cleanup-task"
+	dcResource            = fmt.Sprintf("CassandraDatacenter/%s", dcName)
+	scheduledTaskResource = fmt.Sprintf("ScheduledTask/%s", scheduledTaskName)
+	dcLabel               = fmt.Sprintf("cassandra.datastax.com/datacenter=%s", dcName)
+	ns                    = ginkgo_util.NewWrapper(testName, namespace)
+)
+
+func TestLifecycle(t *testing.T) {
+	AfterSuite(func() {
+		logPath := fmt.Sprintf("%s/aftersuite", ns.LogDir)
+		err := kubectl.DumpAllLogs(logPath).ExecV()
+		if err != nil {
+			t.Logf("Failed to dump all the logs: %v", err)
+		}
+
+		fmt.Printf("\n\tPost-run logs dumped at: %s\n\n", logPath)
+		ns.Terminate()
+		err = kustomize.Undeploy(namespace)
+		if err != nil {
+			t.Logf("Failed to undeploy cass-operator: %v", err)
+		}
+	})
+
+	RegisterFailHandler(Fail)
+	RunSpecs(t, testName)
+}
+
+var _ = Describe(testName, func() {
+	Context("when in a new cluster", func() {
+		Specify("the operator can schedule and execute CassandraTasks", func() {
+			var step string
+			var json string
+			var k kubectl.KCmd
+
+			By("deploy cass-operator with kustomize")
+			err := kustomize.Deploy(namespace)
+			Expect(err).ToNot(HaveOccurred())
+
+			step = "waiting for operator pod to be created"
+			json = "jsonpath={.items[*].metadata.name}"
+			k = kubectl.Get("pods").
+				WithLabel("name=cass-operator").
+				FormatOutput(json)
+			ns.WaitForOutputContainsAndLog(step, k, "cass-operator", 300)
+
+			ns.WaitForOperatorReady()
+
+			step = "creating a datacenter resource with 2 nodes"
+			testFile, err := ginkgo_util.CreateTestFile(dcYaml)
+			Expect(err).ToNot(HaveOccurred())
+
+			k = kubectl.ApplyFiles(testFile)
+			ns.ExecAndLog(step, k)
+
+			ns.WaitForDatacenterReady(dcName)
+
+			step = "creating a ScheduledTask for cleanup"
+
+			k = kubectl.ApplyFiles(scheduledTaskYaml)
+			stdout, stderr, err := ns.ExecVCapture(k)
+			Expect(err).ToNot(HaveOccurred())
+			fmt.Println(stdout)
+			fmt.Println(stderr)
+
+			step = "verify ScheduledTask is created"
+			json = "jsonpath={.metadata.name}"
+			k = kubectl.Get(scheduledTaskResource).FormatOutput(json)
+			ns.WaitForOutputAndLog(step, k, scheduledTaskName, 60)
+
+			step = "verify ScheduledTask status fields are populated"
+			json = "jsonpath={.status.nextSchedule}"
+			k = kubectl.Get(scheduledTaskResource).FormatOutput(json)
+			// Wait for nextSchedule to be set (non-empty)
+			Eventually(func() string {
```

Comment:
ns.WaitForOutputPatternAndLog / WaitForOutputContainsAndLog / etc

Reply:
I find that difficult to understand what is going on, and often those functions swallow errors in a way that makes it hard for me to use them.

Reply:
What they do is catch the state of the cluster in the final logs, which makes it easier to see how the situation developed from cluster perspective (and what operator logged). Doing it this way removes that functionality and you need to grep cass-operator logs and test logs and try to match the timestamps.

Reply:
I'm sorry, I don't understand your point on this, is it a blocking issue?